inputremapper/injection/consumer_control.py

```diff
@@ -93,23 +104,40 @@ async def run(self):
                 # won't appear, no need to forward or map them.
                 continue
 
+            for listener in self.context.listeners:
+                asyncio.create_task(listener(event))
+                # allow the just created task to do some stuff
+                await asyncio.sleep(0)
```

Comment:
can you reuse the complete old `delayed_handle_keycode` function here? Especially so that the docstring with the explanation is preserved